tests/types/test_string_type.cpp

```diff
@@ -455,41 +455,146 @@ TEST(StringType, Concatenation) {
     EXPECT_ARRAY_EQ("twogamma", c(2));
 }
 
-/*
-TEST(StringType, Find1) {
-    nd::array a, b, c;
 
-    const char *a_arr[4] = {"abc", "ababc", "ababab", "abd"};
-    a = a_arr;
-    b = "abc";
-
-    c = a.f("find", b).eval();
-    ASSERT_EQ(ndt::type("4 * intptr"), c.get_type());
-    ASSERT_EQ(4, c.get_shape()[0]);
-    EXPECT_EQ(0, c(0).as<intptr_t>());
-    EXPECT_EQ(2, c(1).as<intptr_t>());
-    EXPECT_EQ(-1, c(2).as<intptr_t>());
-    EXPECT_EQ(-1, c(3).as<intptr_t>());
+TEST(StringType, Find1) {
+  nd::array a, b, c;
+
+  const char *a_arr[4] = {"abc", "ababc", "ababab", "abd"};
+  a = a_arr;
+  b = "abc";
+
+  c = nd::string_find(a, b);
+  ASSERT_EQ(ndt::type("4 * intptr"), c.get_type());
+  ASSERT_EQ(4, c.get_shape()[0]);
+  EXPECT_ARRAY_EQ(0l, c(0));
+  EXPECT_ARRAY_EQ(2l, c(1));
+  EXPECT_ARRAY_EQ(-1l, c(2));
+  EXPECT_ARRAY_EQ(-1l, c(3));
 }
 
```

Comment:
You can match a whole array with `EXPECT_ARRAY_EQ` by passing in the values as an `std::initializer_list`. For instance, this should be equivalent to what you have above: `EXPECT_ARRAY_EQ(nd::array{0l, 2l, -1l, -1l}, nd::string_find(a, b))`. The benefit of that is it does everything you have (checks type, shape, and values), but we get it down to a single line.